jupyter related packages updates 2024-04-07

natsukium opened the pull request:
## Description of changes

Update jupyter-related packages at once to avoid building hundreds of packages over and over again.

This PR is based on the commits generated by the following command.

```
nix search nixpkgs#python311Packages 'ipython|jupyter|notebook' --json | jq -r 'keys | .[]' | cut -d . -f 3,4 | xargs -n 1 nix-update --commit
```

closes https://github.com/NixOS/nixpkgs/pull/299772, closes https://github.com/NixOS/nixpkgs/pull/302312, closes https://github.com/NixOS/nixpkgs/pull/301479, closes https://github.com/NixOS/nixpkgs/pull/302106, closes https://github.com/NixOS/nixpkgs/pull/302273, closes #302425, closes https://github.com/NixOS/nixpkgs/pull/302467, closes  https://github.com/NixOS/nixpkgs/pull/302602, closes https://github.com/NixOS/nixpkgs/pull/302643, closes #303023
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

natsukium committed:
python311Packages.anywidget: 0.9.4 -> 0.9.6

Changelog: https://github.com/manzt/anywidget/releases/tag/anywidget%400.9.6

natsukium committed:
python311Packages.ipykernel: 6.29.3 -> 6.29.4

Changelog: https://github.com/ipython/ipykernel/releases/tag/v6.29.4

natsukium committed:
python311Packages.ipyparallel: 8.7.0 -> 8.8.0

Changelog: https://github.com/ipython/ipyparallel/blob/8.8.0/docs/source/changelog.md

natsukium committed:
python311Packages.ipyvuetify: 1.9.3 -> 1.9.4

natsukium committed:
python311Packages.nbformat: 5.10.3 -> 5.10.4

natsukium committed:
python311Packages.pytest-jupyter: 0.9.1 -> 0.10.1

Diff: https://github.com/jupyter-server/pytest-jupyter/compare/refs/tags/v0.9.1...v0.10.1

Changelog: https://github.com/jupyter-server/pytest-jupyter/releases/tag/v0.10.1

dmadisetti reviewed (approved):
Marimo doesn't depend on ipykernel- it's an alternative. But we'll take the update

natsukium committed:
python311Packages.jupyterlab: 4.1.5 -> 4.1.6

Changelog: https://github.com/jupyterlab/jupyterlab/blob/v4.1.6/CHANGELOG.md

natsukium committed:
python311Packages.jupyterlab-server: 2.25.4 -> 2.26.0

Changelog: https://github.com/jupyterlab/jupyterlab_server/blob/v2.26.0/CHANGELOG.md

natsukium commented:
@dmadisetti Thanks for your checking!
While waiting for nixpkgs-review, marimo is bumped to [a newer version](https://github.com/NixOS/nixpkgs/pull/303062), so I excluded it from this PR.

dmadisetti commented:
Hmm, maybe remove marimo in favor of #303062? live I said, it's decoupled from jupyter

but thanks!

natsukium commented:
There is no regression related to this PR.
